tests/Properties.hs

```diff
@@ -245,6 +246,74 @@ prop_readIntBoundsLC     =     rdWordBounds @Word
 
 ------------------------------------------------------------------------
 
+expectSizeOverflow :: a -> Property
+expectSizeOverflow val = ioProperty $ do
+  isLeft <$> try @P.SizeOverflowException (evaluate val)
+
+prop_checkedAdd = forAll (vectorOf 2 nonNeg) $ \[x, y] -> if oflo x y
+  then expectSizeOverflow (P.checkedAdd "" x y)
+  else property $ P.checkedAdd "" x y == x + y
+  where nonNeg = choose (0, (maxBound @Int))
+        oflo x y = toInteger x + toInteger y /= toInteger (x + y)
```

Comment:
Please use `-XTypeApplications` or type signatures to make the types involved in the `toInteger` conversions more clear.